cores/esp32/HardwareSerial.cpp

```diff
@@ -320,6 +340,12 @@ void HardwareSerial::begin(unsigned long baud, uint32_t config, int8_t rxPin, in
     if (_uart != NULL && (_onReceiveCB != NULL || _onReceiveErrorCB != NULL) && _eventTask == NULL) {
         _createEventTask(this);
     }
+
+    // Set UART RX timeout
+    if (_uart != NULL && _onReceiveCB != NULL) {
+        uart_set_rx_timeout(_uart_nr, _onReceiveTimeout);
+    }
+    
```

Comment:
As I said before. The timeout behaviour is the default as per IDF (10 symbols defined in UART_TOUT_THRESH_DEFAULT). I think this is needed because the user could call setRxTimeout before or after the begin() . This is the same principle as you took with onReceive(). This is because uart_set_rx_timeout should be called once the uart is initialized.

Reply:
OK. I agree. Let's set the default always.